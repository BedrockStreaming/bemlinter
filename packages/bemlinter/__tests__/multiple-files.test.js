const {lint, format} = require('../src/bemlinter.js');

const snap = (fileName, done, options = {}) => {
  lint(`${__dirname}/sources/${fileName}`, options)
    .then(lintResult => format(lintResult, false))
    .then(output => {
      expect(output).toMatchSnapshot();
      done();
    })
    .catch(error => {
      console.error(error);
      done();
    })
  ;
};

describe('Bemlinter of multiple files', () => {
  it('should lint with crossed error', done => snap('cross-styling/*.scss', done, {classPrefix: ['c-']}));
  
  it('should lint without the crossed error on the excluded block', done => snap('cross-styling/*.scss', done, {
    excludeBlock: ['other-block'],
    classPrefix: ['c-']
  }));
});