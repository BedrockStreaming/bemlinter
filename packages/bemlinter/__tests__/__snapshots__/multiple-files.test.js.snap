exports[`Bemlinter of multiple files should lint with crossed error 1`] = `
"  ✗ block
[block.scss:8] Error: \".c-other-block\" is incoherent with the file name.
  ✗ other-block
[block.scss:8] Error: \".c-other-block\" should not be styled outside of its own stylesheet.

FAIL: bemlinter has detected 2 errors on 2 blocks."
`;

exports[`Bemlinter of multiple files should lint without the crossed error on the excluded block 1`] = `
"  ✗ block
[block.scss:8] Error: \".c-other-block\" is incoherent with the file name.

FAIL: bemlinter has detected 1 error on 1 block."
`;
